### ISSUE TITLE
Hypre fails to install with EasyBuild v3.7.0

### ISSUE THREAD

**Issue by boegel** [easybuilders/easybuild-easyconfigs#6922]
`configure` fails with (from `config.log`):

```
configure:2477: checking host system type
configure:2490: result:
configure:2494: error: invalid value of canonical host
```

The problem seems to be the `--build=x86_64-pc-linux-gnu` that now gets passed by the enhanced `ConfigureMake` easyblock (cfr. https://github.com/easybuilders/easybuild-easyblocks/pull/1506)

cc @ocaisa

**Comment by boegel** [comment]
Hmm, it seems like this happens because we're not also setting `--host`.

If I include `configopts = "--host=x86_64-pc-linux-gnu"` into the `Hypre` easyconfig, it fixes the build.

@ocaisa Any idea if we should (always) be setting *both* `--build=x86_64-pc-linux-gnu` *and* `--host=x86_64-pc-linux-gnu` rather than only `--build=x86_64-pc-linux-gnu`?

cc @geimer

**Comment by ocaisa** [comment]
No sorry, I don't know enough about Automake to know that

**Comment by ocaisa** [comment]
I think if we do that then cross-compilation is out but not sure

**Comment by ocaisa** [comment]
Looks like this should be fine... https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html

**Comment by boegel** [comment]
@ocaisa Well, we would only set the default value for `--build` and `--host`, of course we would allow customization via `build_type = '...'` and `host_type = '...'` in the easyconfig file, so it doesn't prevent cross-compilation at all imho.

**Comment by vanzod** [comment]
I think the solution proposed in https://github.com/easybuilders/easybuild-easyblocks/pull/1531 is elegant and functional enough. I tested it with the new easyconfig in #6923 and it works. I then went ahead and merged it.
Thanks @boegel

**Comment by boegel** [comment]
@vanzod Well, the `build_type = False` workaround isn't more than that, a workaround.

I think it's actually a sign of a more common issue, i.e. that we should also specify `--host` rather than only `--build`...

If we fix that, we don't need to change the `Hypre` easyconfigs at all (and potentially also won't have to touch others).

**Comment by micaeljtoliveira** [comment]
@boegel This is definitely not a common issue. The problem comes from the way how the `Hypre` `configure.in` file is written:
```
dnl *********************************************************************
dnl * Determine BUILD, HOST, and TARGET types
dnl *********************************************************************
if test "x$build_alias" = "x"
then
   AC_CANONICAL_BUILD
fi

if test "x$host_alias" = "x"
then
   AC_CANONICAL_HOST
fi
```
If I understand correctly, what happens is the following: if `--build` is set, then `AC_CANONICAL_BUILD` is not called and the `build` variable is not set. Next, since `build` is not set, the call to `AC_CANONICAL_HOST` fails, as `build` is the default value when `--host` is not defined. See https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Canonicalizing.html for more details. Therefore I would say this is a bug in the `Hypre` build system, because both `AC_CANONICAL_HOST` and `AC_CANONICAL_BUILD` macros should always be called.

**Comment by boegel** [comment]
Thanks for the feedback on this @micaeljtoliveira! That was indeed my take on it too, it's basically `Hypre` getting it wrong there, and the `configure` script tripping over only setting `--build` (which is now fixed with https://github.com/easybuilders/easybuild-easyblocks/pull/1532).

I've verified that setting both `--build` and `--host` fixes the `Hypre` build, so I'm closing this (and also https://github.com/easybuilders/easybuild-easyconfigs/pull/6923 which is an inferior workaround anyway).
